skl2onnx/algebra/onnx_operator.py

```diff
@@ -0,0 +1,357 @@
+    :param domain: to overwrite the default domain
+    :param kwargs: additional parameters of the operator
+    """
+    class OnnxOperatorVariable:
+
+        def __init__(self, index, name=None):
+            self.index = index
+            self.name = name
+
+        def __repr__(self):
+            return "OnnxOperatorVariable('%s')" % self.name
+
+    class UnscopedVariable:
+        def __init__(self, name):
+            self.name = name
+
+        def __eq__(self, name):
+            if isinstance(name, str):
+                return name == self.name
+            elif isinstance(name, OnnxOperator.UnscopedVariable):
+                return self.name == name.name
+            else:
+                raise TypeError('Unsupported type for comparison {}'.format(
+                    type(name)))
+
+        def __repr__(self):
+            return "UnscopedVariable('%s')" % self.name
+
+    class ConstantVariable:
+        def __init__(self, value):
+            self.value = value
+
+        @property
+        def ConstantValue(self):
+            return self.value
+
+    def __init__(self, *inputs, op_version=None, output_names=None,
+                 domain=None, **kwargs):
+        self.state = None
+        self.op_version = op_version or get_opset_number_from_onnx()
+        self.domain = domain
+        self.kwargs = kwargs
+
+        # check inputs
+        if len(inputs) == 0:
+            if self.input_range[0] == self.input_range[1]:
+                self.inputs = [_[0] for _ in self.__class__.expected_inputs]
+            else:
+                # The number of inputs may vary.
+                self.inputs = None
+        else:
+            self.inputs = []
+            for inp in inputs:
+                if isinstance(inp, str):
+                    self.inputs.append(OnnxOperator.UnscopedVariable(inp))
+                elif isinstance(inp, (OnnxOperator, Variable)):
+                    self.inputs.append(inp)
+                elif isinstance(inp, (np.ndarray, TensorProto)):
+                    self.inputs.append(OnnxOperator.ConstantVariable(inp))
+                elif isinstance(inp, OnnxOperator.OnnxOperatorVariable):
+                    self.inputs.append(inp)
+                else:
+                    raise TypeError("Unable to interpret the "
+                                    "input name for type {}.".format(
+                                        type(inp)))
+
+        if self.inputs is not None:
+            if len(self.inputs) < self.input_range[0] or \
+                    len(self.inputs) > self.input_range[1]:
+                raise RuntimeError("Operator '{}' expects a number of inputs "
+                                   "in [{}, {}] not {}".format(
+                                       self.operator_name,
+                                       *self.input_range,
+                                       len(self.inputs)))
+
+        # check output
+        if hasattr(output_names, 'outputs') and \
```

Comment:
() are preferred over `\`.

Reply:
Same here, flake8 should be there for that.